File: 62750938-main/credit/credit.c
#include <cs50.h>
#include <stdio.h>


int main(void)
{
    long number = get_long("Number: ");
    int digit = 0;
    int code = 0;
    bool odd = true;
    int initials = 0;
    int digitscount = 0;


    while (number > 0)
    {
        //selecting the last digit
        digit = number % 10;


        if(odd == false){
            digit = digit * 2;

            //Check if the double of the number is bigger than 10, and if it is sum both the algarisms
            if(digit < 10)
            {
                code += digit;
            }
            else
            {
                code += (int) digit / 10;
                code += digit % 10;
            }
            odd = true;
        }
        else
        {
            code += digit;
            odd = false;
        }

        //deleting the last digit from the number
        number = number / 10;

        if(number < 100 && number >= 10)
        {
            initials = number;
        }

        //counting the digits to check with the initials
        digitscount++;
    }


    //Checking if the checksum ends in 0
    if(code % 10 != 0)
    {
        printf("INVALID\n");
        return 0;
    }

    if((initials == 34 || initials == 37) && digitscount == 15)
    {
        printf("AMEX\n");
    }
    else if((initials < 56 && initials > 50) && digitscount == 16)
    {
        printf("MASTERCARD\n");
    }
    else if(initials / 10 == 4 && (digitscount == 13 || digitscount == 16))
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

File: 62750938-main/sentimental-credit/credit.py
# Create a check for the input
while True:
    try:
        number = int(input("Number: "))
        if number > 0:
            break

    except ValueError:
        pass

odd = True
checksum = 0
initials = 0
digitscount = 0

while number > 0:
    digit = number % 10

    if not odd:
        digit *= 2

        if digit < 10:
            checksum += digit
        else:
            checksum += digit // 10 + digit % 10

        odd = True
    else:
        checksum += digit
        odd = False

    number //= 10  # Use integer division

    if number >= 10 and number < 100:
        initials = number

    digitscount += 1


if checksum % 10 != 0:
    print("INVALID")
else:
    if (initials == 34 or initials == 37) and digitscount == 15:
        print("AMEX")
    elif (initials < 56 and initials > 50) and digitscount == 16:
        print("MASTERCARD")
    elif initials // 10 == 4 and (digitscount == 13 or digitscount == 16):
        print("VISA")
    else:
        print("INVALID")
